packages/flutter/lib/src/rendering/animated_size.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/animation.dart';
+
+import 'box.dart';
+import 'object.dart';
+import 'shifted_box.dart';
+
+/// A render object that can render child size change animations.
+class RenderAnimatedSize extends RenderAligningShiftedBox {
+  /// Creates a render animated size.
+  RenderAnimatedSize({
+    Duration duration,
+    Curve curve,
+    RenderBox child,
+    FractionalOffset alignment: FractionalOffset.center,
+    VoidCallback callback
+  }) : super(child: child, alignment: alignment) {
+    _controller = new AnimationController(
+      duration: duration
+    );
+    _animation = new CurvedAnimation(
+      parent: _controller,
+      curve: curve
+    );
+    _callback = callback;
+  }
+
+  AnimationController _controller;
+  CurvedAnimation _animation;
+  Size _sourceSize;
+  Size _targetSize;
+  bool _lastFrameChanged = false;
+  bool _hasVisualOverflow;
+
+  VoidCallback get callback => _callback;
+  VoidCallback _callback;
+  set callback(VoidCallback value) {
+    if (value == _callback)
+      return;
+    _callback = value;
+  }
+
+  /// The duration of the animation.
+  Duration get duration => _controller.duration;
+  set duration(Duration value) {
+    assert(value != null);
+    if (value == _controller.duration)
+      return;
+    _controller.duration = value;
+  }
+
+  /// The curve of the animation.
+  Curve get curve => _animation.curve;
+  set curve(Curve value) {
+    assert(value != null);
+    if (value == _animation.curve)
+      return;
+    _animation.curve = value;
+  }
+
+  @override
+  void detach() {
+    _controller?.stop();
+    super.detach();
+  }
+
+  void _trigger() {
+    _controller
+      ..removeListener(markNeedsLayout)
+      ..value = 0.0
+      ..forward()
+      ..addListener(markNeedsLayout);
+  }
+
+  Size get _animatedSize {
+    return Size.lerp(_sourceSize, _targetSize, _animation.value);
+  }
+
+  @override
+  void performLayout() {
+    if (_callback != null)
+      _callback();
+
+    _hasVisualOverflow = false;
+
+    if (child != null) {
+      child.layout(constraints, parentUsesSize: true);
+      final Size childSize = constraints.constrain(child.size);
+
+      if (_targetSize != childSize) {
+        _sourceSize = _animatedSize ?? childSize;
+        _targetSize = childSize;
+
+        if (_lastFrameChanged) {
```

Comment:
I'd rename this to `_didChangeTargetSizeLastFrame` to make these if statements read better "if the object did change target size last frame" is closer to english grammar than "if the object last frame changed"